README.md
# Nuxt 3 + Tailwind CSS + DaisyUI Website
[![Netlify Status](https://api.netlify.com/api/v1/badges/1fcf26af-7f44-4e56-abe5-94ae4073f109/deploy-status)](https://app.netlify.com/sites/requiem-blog/deploys)

Personal blog made by [Nuxt 3](https://nuxt.com/)
Click [Link](https://requiem-blog.netlify.app/) to see the page

## Setup
### Install dependencies
```bash
yarn install
````

### Development
```bash
yarn dev
```
The website will run on port 3005

### Production
Build the application for production:
```bash
yarn build
```

### Preview build:
```bash
yarn preview
```

## config
### .env
```text
NUXT_PUBLIC_GOOGLE_ANALYTICS_ID="GOOGLE ANALYTICS KEY"
NUXT_PUBLIC_SITE_URL="PUBLIC URL"
NUXT_PUBLIC_CA_PUB="GOOGLE CONSOLE KEY"
NUXT_PUBLIC_AMZN_ASSOC_TARCKING_ID="AMAZON ASSOCIATE TRAKING ID"
```
### app.config.ts
| Name                 | Description         |
|----------------------|---------------------|
| github               | github links        |
| blogCategories       | blog categories     |
| blogTags.programming | Programming tags    |
| showcaseCategories   | Showcase categories |
| showcaseTags         | Showcase tags       |


## Packages
### Nuxt Modules
- [@nuxt/content](https://content.nuxtjs.org/): Content made easy for Vue Developers
- [@pinia/nuxt](https://pinia.vuejs.org/ssr/nuxt.html) - Pinia module for Nuxt
  - [pinia](https://nuxt.com/modules/pinia) - Type Safe, Extensible, and Modular by design. Forget you are even using a store.
- [@nuxtjs/i18n](https://github.com/nuxt-modules/i18n) - vueI18n module for Nuxt 3
- [@vueuse/nuxt]() - vueuse module for Nuxt 3
- [@nuxtjs/tailwindcss](https://tailwindcss.nuxt.dev/) - Tailwind module for Nuxt 3
- [@nuxtjs/color-mode](https://color-mode.nuxtjs.org/) - Dark and Light mode with auto detection made easy with Nuxt
### Relative with Content
- [highlight.js]() - Add highlight for code section
- [rehype-highlight]() - Add highlight for code section (not working)
- [remark-reading-time](https://www.npmjs.com/package/remark-reading-time) - Add reading time
### ETC
- [tailwindcss](https://tailwindcss.com/)= Tailwind css
  - [postcss-custom-properties](https://www.npmjs.com/package/postcss-custom-properties) - Fix ```nuxt Cannot start nuxt: Cannot find module 'postcss-custom-properties'``` error
- [daisyui](https://daisyui.com/) - Tailwind based UI Framework
- [theme-change](https://github.com/saadeghi/theme-change) - Change theme with daisyui
- [dayjs](https://day.js.org/): 
- [fast-glob]() - Find all files under specific directory
- ~~[vue-gtag]()~~
- [vue-gtag-next]()
- [animate.css](https://animate.style/) - A library of ready-to-use, cross-browser animations
- [sitemap](https://github.com/ekalinin/sitemap.js) - Generate sitemap
- ~~[vite-pwa](https://github.com/vite-pwa/nuxt)~~
- [consola](https://github.com/unjs/consola) - Fix the error of consola

## :file_folder: Folder structure
See more detail to click the [Link](https://nuxt.com/docs/guide/directory-structure/nuxt)
```text
├─ assets/                      # Assets (Videos, fonts ...)
├─ components/                  # Components
├─ content/                     # Content ( .md, .mdx ...)
├─ layouts/                     # Layouts
├─ locales/                     # i18n locales files
├─ node_modules/                # node modules
├─ pages/                       # Pages
├─ plugins/                     # Vue Plugins (Server, Client), Automatically adjust
├─ public/                      # Assets (images ...)
├─ server/                      # Server files
│  ├─ api/                      # API, File name will be path
│  ├─ public/                   # Data for server
│  ├─ routes/                   # Generate page from serverside
├─ .eslintrc.cjs                 # Eslint config.
├─ .gitignore                   # gitignore.
├─ nitro.config.ts              # Nitro build config.
├─ app.vue                      # App vue file
├─ app.config.ts                # App config
├─ nuxt.config.ts               # Nuxt 3 config
├─ package.json                 # Package.json
├─ README.md                    # README.md
├─ tailwind.config.js           # Tailwind CSS config.
├─ tsconfig.json                # Typescript config
```

## Contributors
Thank you for helping me to make better blog.

[//]: # (max 7 td in each tr)
[//]: # (<a href="https://github.com/kkan0615/blog-nuxt/commits?author=kkan0615" title="Examples">💡</a> )
[//]: # (<a href="https://github.com/kkan0615/blog-nuxt/commits?author=kkan0615" title="Tests">⚠️</a>)
[//]: # (<a href="https://github.com/kkan0615/blog-nuxt/commits?author=kkan0615" title="Ideas, Planning, & Feedback">🤔</a>)
[//]: # (<a href="https://github.com/kkan0615/blog-nuxt/issues?q=author%3Akkan0615" title="Bug reports">🐛</a>)
<table>
  <tbody>
    <tr>
      <td align="center">
        <a href="https://github.com/kkan0615">
          <img src="https://avatars.githubusercontent.com/u/46660361?v=4?s=64" width="64px;" alt="Youngjin Kwak"/><br /><sub><b>Youngjin Kwak</b></sub>
        </a><br />
        <a href="https://github.com/kkan0615/blog-nuxt/commits?author=kkan0615" title="Code">💻</a> 
        <a href="https://github.com/kkan0615/blog-nuxt/commits?author=kkan0615" title="Maintenance">🚧</a>
        <a href="https://github.com/kkan0615/blog-nuxt/commits?author=kkan0615" title="Documentation">📖</a> 
        <a href="https://github.com/kkan0615/blog-nuxt/commits?author=kkan0615" title="Design">🎨</a>
      </td>
    </tr>
  </tbody>
</table>

## Supports
[!["Buy Me A Coffee"](https://www.buymeacoffee.com/assets/img/custom_images/orange_img.png)](https://www.buymeacoffee.com/youngjinkwak)

<img src="./public/donations/paypal-qrcode.png" alt="paypal" width="250" height="250">

[![](https://c5.patreon.com/external/logo/become_a_patron_button.png)](https://www.patreon.com/youngjin_kwak)
## Reference Sites
- [Nuxt 3 documentation](https://nuxt.com/docs/getting-started/introduction)
- [deployment documentation](https://nuxt.com/docs/getting-started/deployment)
- [Github - Nuxt](https://github.com/nuxt/nuxt)
- [Nuxt Tailwind with DaisyUI and color mode.](https://tailwindcss.nuxtjs.org/examples/daisyui)


CHANGELOG.md
## 1.0.2
- Added new content
- Delete highlights

## 1.0.0
- Update to new version

## 0.3.0
### Core Changes
- Added new tags (php and laravel)
- Update nuxt version
- Update nuxt/i18n version
### Documentation Changes
- Added new content "Javascript - generate tree from Array"
- Added new content "Laravel - handling columns and table in migration"
- Added new content "Javascript - generate tree from Array"

## 0.2.9
### Core Changes
- Update donation button
- Update layout
### Documentation Changes
- Added new content "Select all text in input on focused"

## 0.2.8
### Core Changes
- Updated pagination component
- Fixed undefined issue.
### Documentation Changes
- Added new content "Select all text in input on focused"
- Added new content "Typescript Array Add, Update, Delete elements"
- Updated date in content "JavaScript 5 ways to deep copy objects"

## 0.2.7s
### Documentation Changes
- Added new content "JavaScript console.time()"
- Added new content "JavaScript 5 ways to deep copy objects"

## 0.2.6
### Core Changes
- Fix i18n error
- Updated all packages to latest

## 0.2.5
### Core Changes
- Fixed locale errors
### Documentation Changes
- Added new content "JavaScript Destructing assignment"
- Added new content "JJavaScript at() method"
### Misc Changes
- Change TOC component name to TableOfContent 
- Hide "example" in categories

## 0.2.4
### Core Changes
- Updated all packages to latest
- Updated design based on Daisy.UI 3.0
### Documentation Changes
- Added new content "Fullscreen API in Typescript"
### Misc Changes
- Cookie privacy page is created, but not done yet

## 0.2.3
### Core Changes
- Updated version of Nuxt i18n
### Documentation Changes
- Added new content "Javascript FormData Object"
- Added new content "File System Access API with Typescript"

## 0.2.2
### Core Changes
- Increased font-size for posts
- Updated version of content
- Updated version of Nuxt
- Updated version of nuxt/typescript
- Updated version of nuxt/pinia
### Documentation Changes
- Added new content "Javascript FormData Object"
- Added new content "Next.js 13 app dir Route handler with Typescript"
### Misc Changes
- Added open-collective in funding.yml
- Added next-logo.jpeg

## 0.2.1
### Core Changes
- Updated version of nuxt/tailwindcss
- Updated theme select component
- Added nuxt/color-mode

## 0.2.0
### Core Changes
- Fixed the error to redirect to blog when you click the share button on card
- Removed feature to add category when you click the blog 
- Updated version of nuxt
### Documentation Changes
- Added new content "Two ways to connect WebSocket in Vue 3"
### Misc Changes
- Added paypal donation in README.md 

## 0.1.9
### Core Changes
- Update google console log

## 0.1.8
### Core Changes
- Added animation for dialog
- Updated Donation Modal
- Updated Dialog backdrop
- Updated share button position in blog card component
- Updated share button component order of elements
- Updated colors: secondary and accent
- Remove background color of back button component
- Fixed error for color of sidebar in layout on mobile
### Documentation Changes
- Added new content "CSS any-link pseudo-class selector"

## 0.1.7 (04-09-2023)
### Core Changes
- Updated share button
- Added share button in blog
- Updated round of card image
- Updated color pallet
- Update home style
- Update style of navbar on blogs and showcases pages
- Updated layout styles
### Documentation Changes
- Added new content "Typescript Enum vs const Enum"
### Misc Changes
- Added funding
- Added and Updated i18n labels

Check release for previous versions....


app.config.ts
export default defineAppConfig({
  startedAt: '2023-01-13',
  github : 'https://github.com/kkan0615/blog-nuxt',
  blogCategories: [
    // 'example',
    'programming',
  ],
  blogTags: {
    programming: [
      'web',
      'node Js',
      'get Started',
      'javaScript',
      'typescript',
      'css',
      'react',
      'vue',
      'next',
      'nuxt',
      'MDX',
      'HTML',
      'php',
      'laravel',
    ]
  },
  showcaseCategories: [
    'example',
    'desktopApp',
    'web',
    'mobileApp',
  ],
  showcaseTags: [
    'node Js',
    'typescript',
    'javaScript',
    'next',
    'nuxt',
    'react',
    'vue',
    'electron'
  ],
  profile: {
    image: 'https://avatars.githubusercontent.com/u/46660361?v=4',
    email: 'kkan0615@gmail.com',
    socials: {
      linkedIn: 'https://www.linkedin.com/in/youngjin-kwak-91658323a/',
      indeed: 'https://profile.indeed.com/?hl=en_CA&co=CA&from=gnav-homepage&_ga=2.229680351.2145071738.1672341480-1054300716.1666887722',
      github: 'https://github.com/kkan0615',
    }
  },
})
